persistence/javadsl/src/main/java/com/lightbend/lagom/javadsl/persistence/PersistentEntityRef.java

```diff
@@ -74,6 +75,24 @@ public String entityId() {
         Patterns.ask(region, new CommandEnvelope(entityId, command), timeout);
 
     return future
+        .exceptionally(
+            cause -> {
+              if (cause instanceof AskTimeoutException) {
+                String msg =
+                    "Ask timed out on ["
+                        + this
+                        + "] after ["
+                        + timeout.toMillis()
+                        + " ms]. Message of type ["
+                        + command.getClass()
+                        + "]. "
+                        + "A typical reason for `AskTimeoutException` is that the recipient actor didn't send a reply.";
+                return new AskTimeoutException(msg, cause);
```

Comment:
This is odd, `CompletionStage` doesn't have an API to recover with another `CompletionStage` so we need to return an `Exception` which will create a `CompletionStage<Throwable>` that we will later convert to a failed one. See `thenCompose` after it.